specification/dartLangSpec.tex

```diff
@@ -11267,33 +11267,24 @@ \subsection{Function Expressions}
 which is used below and in other sections, as follows:
 
 \begin{itemize}
-\item If $T$ is \code{FutureOr<$S$>} for some $S$ then $\flatten{T} = S$.
+\item If $T$ is a type variable $X$ with bound $B$,
```

Comment:
OK, so we're updating flatten anyway! But in that case we would basically have to assume the null safety specification as the context, because the new definition of flatten includes `S?` as a type. This is a little bit inconvenient because we don't have all the sections (that is, `\ref{X}` is needed for some `X` that don't exist), but I guess it should be OK to have a spec with unresolved refs for a while.

In any case, we'll land the null safety update any day now, of course! ;-)

Reply:
Of course, of course.

I decided to omit the `S?` case because I assumed it would be merged in with the rest of null safety,
so you only needed to merge the one line here into the patch.

Also why I didn't use `\DefEquals`, since the other existing cases didn't, and changing those would make rebasing the null safety PRs harder.